log.lua

```diff
@@ -15,51 +15,59 @@ for i, level in ipairs(loglevels) do revlevels[level] = i end
 
 local startuploglevel = os and os.getenv
                            and revlevels[string.upper(os.getenv("LUALOG") or "")]
--- TODO: provide interfaces to dynamically set log level
+
 local currentloglevel = startuploglevel
 
-local function defaultlogger(from, level, ...)
+local function defaultlogger(opts, ...)
   if not currentloglevel then return end
-  if level >= currentloglevel then
+  if opts.level >= currentloglevel then
     local components = {}
     for _, v in ipairs({...}) do table.insert(components,tostring(v)) end
 
     print(string.format(
       "[%s]%s %s",
-      loglevels[level],
-      from and "("..from..")" or "",
+      loglevels[opts.level],
+      opts.from and "("..opts.from..")" or "",
       table.concat(components, "\t")
     ))
   end
 end
 local currentlogger = defaultlogger -- TODO: allow overriding
 
 function log.print(...)
-  currentlogger(nil, revlevels.PRINT, ...)
+  currentlogger({level = revlevels.PRINT}, ...)
 end
 
 function log.fatal(...)
-  currentlogger(nil, revlevels.FATAL, ...)
+  currentlogger({level = revlevels.FATAL}, ...)
 end
 
 function log.error(...)
-  currentlogger(nil, revlevels.ERROR, ...)
+  currentlogger({level = revlevels.ERROR}, ...)
 end
 
 function log.warn(...)
-  currentlogger(nil, revlevels.WARN, ...)
+  currentlogger({level = revlevels.WARN}, ...)
 end
 
 function log.info(...)
-  currentlogger(nil, revlevels.INFO, ...)
+  currentlogger({level = revlevels.INFO}, ...)
 end
 
 function log.debug(...)
-  currentlogger(nil, revlevels.DEBUG, ...)
+  currentlogger({level = revlevels.DEBUG}, ...)
 end
 
 function log.trace(...)
-  currentlogger(nil, revlevels.TRACE, ...)
+  currentlogger({level = revlevels.TRACE}, ...)
+end
+
+function log.setLogLevel(level)
+  currentloglevel = level
+end
```

Comment:
I'm kinda thinking this should be a feature of the logger. Or it should have some way to communicate to the configured logger or something.

I think in this PR, any non-default logger wouldn't have any way to know the current log level.

Thoughts?

Reply:
I've updated the default logger to capture its own environment, including the log level. Let me know if you wanted this to happen some other way